ArduCopter/AP_Arming.cpp

```diff
@@ -441,11 +393,13 @@ bool AP_Arming_Copter::gps_checks(bool display_failure)
     // warn about hdop separately - to prevent user confusion with no gps lock
     if (copter.gps.get_hdop() > copter.g.gps_hdop_good) {
         check_failed(ARMING_CHECK_GPS, display_failure, "High GPS HDOP");
+        AP_Notify::flags.pre_arm_gps_check = false;
```

Comment:
I wonder if this could go into check_failed; the first field should be enough to switch on or whatever.

Reply:
@peterbarker, the problem is that it needs to be set to true some times as well and I'd rather not spread that logic out over too many functions.